src/Symfony/Component/Workflow/DataCollector/WorkflowDataCollector.php

```diff
@@ -163,6 +141,41 @@ private function getEventListeners(WorkflowInterface $workflow): array
         return $listeners;
     }
 
+    public function buildMermaidLink(string $name): string
+    {
+        $payload = [
+            'code' => $this->data['workflows'][$name]['dump'],
+            'mermaid' => '{"theme": "default"}',
+            'autoSync' => false,
+        ];
+
+        $compressed = zlib_encode(json_encode($payload), ZLIB_ENCODING_DEFLATE);
+
+        $suffix = rtrim(strtr(base64_encode($compressed), '+/', '-_'), '=');
+
+        return "https://mermaid.live/edit#pako:{$suffix}";
+    }
+
+    protected function getCasters(): array
```

Comment:
why moving this method ? If this is about the order of methods by visibility, the root cause is actually that you are adding your public `buildMermaidLink` in the middle of private methods.